change/@ni-nimble-components-56889b4d-259c-4017-b55f-97248f5f4db3.json

```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "Fix for selected option display text being updated in DOM not being reflected in display of select.",
```

Comment:
Did you test whether or not filtering correctly handles the text changing in an option? I'm guessing not. If it is broken and it doesn't seem like it could be easily fixed within this PR, we should file a separate issue for it.

Reply:
It does. Ultimately when you filter it will re-query the `textContent` for each option, which will always retrieve what's in the DOM.